tests/compile-fail-fullmir/stacked_borrows/invalidate_against_barrier2.rs

```diff
@@ -0,0 +1,13 @@
+fn inner(x: *mut i32, _y: &i32) {
+    // If `x` and `y` alias, retagging is fine with this... but we really
```

Comment:
> Shouldn't this simply invalidate y?

Yes. But without the barrier, invalidating `y` would be okay. Nobody is using `y` again, after all.

Reply:
So why are we disallowing this code? `y` is unused, even in the main function.

Reply:
Because we want to enforce that a reference passed to a function is not invalidated while the function executes. This is an operational version of "the reference outlives the function".